src/test/java/ai/rev/speechtotext/unit/RevAiStreamingClientOptionalParametersTest.java

```diff
@@ -27,103 +27,115 @@
+    private static StreamingClient streamingClient;
+    private SessionConfig sessionConfig = new SessionConfig();
 
-  public RevAiStreamingClientOptionalParametersTest(
-      String metadata, String customVocabularyId, Boolean filterProfanity) {
-    sessionConfig.setMetaData(metadata);
-    sessionConfig.setFilterProfanity(filterProfanity);
-    sessionConfig.setCustomVocabularyId(customVocabularyId);
+    public RevAiStreamingClientOptionalParametersTest(
+            String metadata, String customVocabularyId, Boolean filterProfanity, Boolean removeDisfluencies) {
+        sessionConfig.setMetaData(metadata);
+        sessionConfig.setFilterProfanity(filterProfanity);
+        sessionConfig.setCustomVocabularyId(customVocabularyId);
+        sessionConfig.setRemoveDisfluencies(removeDisfluencies);
 
-    this.defaultContentType = new StreamContentType();
-    defaultContentType.setContentType("audio/x-raw");
-    defaultContentType.setLayout("interleaved");
-    defaultContentType.setFormat("S16LE");
-    defaultContentType.setRate(16000);
-    defaultContentType.setChannels(1);
-  }
+        this.defaultContentType = new StreamContentType();
+        defaultContentType.setContentType("audio/x-raw");
+        defaultContentType.setLayout("interleaved");
+        defaultContentType.setFormat("S16LE");
+        defaultContentType.setRate(16000);
+        defaultContentType.setChannels(1);
+    }
 
-  @Parameterized.Parameters
-  public static Collection input() {
-    return Arrays.asList(
-        new Object[][] {
-          {METADATA, VOCAB_ID, true},
-          {METADATA, VOCAB_ID, false},
-          {METADATA, null, true},
-          {METADATA, VOCAB_ID, null},
-          {null, VOCAB_ID, true},
-          {null, VOCAB_ID, false},
-          {null, VOCAB_ID, null},
-          {null, null, true},
-          {null, null, false},
-          {null, null, null}
-        });
-  }
+    @Parameterized.Parameters
+    public static Collection input() {
+        return Arrays.asList(
+                new Object[][]{
+                        {METADATA, VOCAB_ID, true, true},
+                        {METADATA, VOCAB_ID, false, true},
+                        {METADATA, VOCAB_ID, null, true},
+                        {METADATA, VOCAB_ID, true, false},
+                        {METADATA, VOCAB_ID, false, false},
+                        {METADATA, VOCAB_ID, null, false},
+                        {METADATA, VOCAB_ID, true, null},
+                        {METADATA, VOCAB_ID, false, null},
+                        {METADATA, VOCAB_ID, null, null},
+                        {METADATA, null, true, true},
+                        {null, VOCAB_ID, true, false},
+                        {null, VOCAB_ID, false, null},
+                        {null, VOCAB_ID, null, true},
+                        {null, null, true, true},
+                        {null, null, false, false},
+                        {null, null, null, null}
```

Comment:
Please check if there's another way to specify these option sets, this matrix is not sustainable. If possible we should change it starting from this PR.

Reply:
☝️ I agree

Reply:
I changed it to `@Theory` and it tests all the cases.